wazo_router_confd/services/carrier.py

```diff
@@ -3,23 +3,40 @@
 
 from sqlalchemy.orm import Session
 
+from wazo_router_confd.auth import Principal
 from wazo_router_confd.models.carrier import Carrier
 from wazo_router_confd.schemas import carrier as schema
+from wazo_router_confd.services import tenant as tenant_service
 
 
-def get_carrier(db: Session, carrier_id: int) -> Carrier:
-    return db.query(Carrier).filter(Carrier.id == carrier_id).first()
+def get_carrier(db: Session, principal: Principal, carrier_id: int) -> Carrier:
+    db_carrier = db.query(Carrier).filter(Carrier.id == carrier_id)
+    if principal is not None and principal.tenant_uuid:
```

Comment:
This is a follow-up of the comment in the middleware where None would be returned when GET `/token/<token>` returns 404.
It would basically skip the tenant segregation part of the query making the unauthenticated request very privileged.

Reply:
This is incorrect, the `auth_middleware` returns 401 (it was 403 before) if principal is None:

```
            principal = await auth_client.get_token_data(header, tenant_uuid)
            if principal is None:
                return Response("The provided token is not valid", status_code=401)
```